### PR TITLE
Update chemfiles to 0.10.0

### PR DESCRIPTION
**Pull request by Luthaf**
Do not merge this yet, it contains a pre-release version to check everything is working fine with Yggdrasil infrastructure.

### CODE REVIEW

File: C/Chemfiles/build_tarballs.jl [hunk 1]
```diff
@@ -3,13 +3,13 @@
 using BinaryBuilder, Pkg
 
 name = "Chemfiles"
-version = v"0.9.3"
+version = v"0.10.0-rc1"
```
Comment (giordano) [inline C/Chemfiles/build_tarballs.jl:6]: Please don't use prerelease versions here, they create some troubles to the package manager

Comment (Luthaf) [reply C/Chemfiles/build_tarballs.jl:6]: As mentioned in the first comment of the PR, this is only a WIP PR to check that everything builds. I'll change the version to the full release before asking for a merge 😃

**Comment by Luthaf** [comment]
This should be good to go!

**Comment by Luthaf** [comment]
thanks!

**Pull request merged**